### ISSUE TITLE
API client app crashes if AnkiDroid does not have storage permissions

### ISSUE THREAD

**Issue by cassdeckard** [ankidroid/Anki-Android#5393]
###### Reproduction Steps

1. Install AnkiDroid but do not run OR remove Storage permissions from existing AnkiDroid app via App Info UI
2. Launch [API Sample app](https://github.com/ankidroid/apisample)
3. Select entry, hit share button, select AnkiDroid Instant-Add

###### Expected Result

Not sure. Maybe user gets prompted with dialog explaining why Anki cannot be exported to?

###### Actual Result

API Sample App crashes due to unhandled `IllegalStateException` in `AnkiDroidHelper::getDeckId` with message "AnkiDroid database inaccessible. Open AnkiDroid to see what's wrong."

###### Debug info
N/A (crash is in the app using the AnkiDroid API, not AnkiDroid itself)

###### Research

[ x ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid

[ x ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue

[ x ] I have searched for similar existing issues here and on the user forum

###### Additional Note

If #5304 were implemented, this issue would likely go away.

**Comment by mattdeckard** [comment]
I submitted this as an [issue ](https://github.com/ankidroid/apisample/issues/10) on the sample app, but the fix may need to happen in the API itself unless there is an elegant way for API clients to avoid this crash (in which case it should probably be documented).

This was discovered because it happens in the Wannichou app ([link to issue](https://github.com/Limegrass/Wanicchou/issues/48)) which exports cards to AnkiDroid.

Options we discussed in that linked thread are:

- Add an API to check to see if AnkiDroid has access to the DB, so clients can use this before attempting to use API calls that would raise exceptions
- Add an API to request storage permission on behalf of AnkiDroid (not sure if this is even possible)

Another option that just occurred to me would be to have the API code catch the exception and handle it by returning null, since that seems to be how most of the API methods report errors.
